edit-post/store/selectors.js

```diff
@@ -40,14 +40,34 @@ export function isPluginSidebarOpened( state ) {
 }
 
 /**
- * Returns the current active general sidebar name.
+ * Returns the current active general sidebar name, or null if there is no
+ * general sidebar active. The active general sidebar is a unique name to
+ * identify either an editor or plugin sidebar.
+ *
+ * Examples:
+ *
+ *  - `edit-post/document`
+ *  - `my-plugin/insert-image-sidebar`
  *
  * @param {Object} state Global application state.
  *
  * @return {?string} Active general sidebar name.
  */
 export function getActiveGeneralSidebarName( state ) {
-	return getPreference( state, 'activeGeneralSidebar', null );
+	// Dismissal takes precedent.
+	const isDismissed = getPreference( state, 'isGeneralSidebarDismissed', false );
+	if ( isDismissed ) {
+		return null;
+	}
+
+	// If not dismissed and an active sidebar is assigned, used.
+	const { activeGeneralSidebar } = state;
+	if ( activeGeneralSidebar !== null ) {
+		return activeGeneralSidebar;
+	}
+
+	// Default to Document tab.
+	return 'edit-post/document';
```

Comment:
Is this necessary here? can't we assign it as a default value in the reducer?

Reply:
>Is this necessary here? can't we assign it as a default value in the reducer?

We can, and we should. Also as a top-level documented constant.